Mathlib/Algebra/Lie/Semisimple/Lemmas.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2025 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import Mathlib.Algebra.Lie.LieTheorem
+import Mathlib.Algebra.Lie.Semisimple.Basic
+
+/-!
+# Lemmas about semisimple Lie algebras
+
+This file is a home for lemmas about semisimple and reductive Lie algebras.
+
+## Main definitions / results:
+ * `LieAlgebra.hasCentralRadical_and_of_isIrreducible_of_isFaithful`: a finite-dimensional Lie
+   algebra with a irreducible faithful finite-dimensional representation is reductive.
+ * `LieAlgebra.hasTrivialRadical_and_of_isIrreducible_of_isFaithful`: a finite-dimensional Lie
+   algebra with a irreducible faithful finite-dimensional trace-free representation is semisimple.
+
+-/
+
+namespace LieAlgebra
+
+open LieModule LieSubmodule Module
+
+variable (k L M : Type*) [Field k] [CharZero k]
+  [LieRing L] [LieAlgebra k L] [Module.Finite k L]
+  [AddCommGroup M] [Module k M] [LieRingModule L M] [LieModule k L M] [Module.Finite k M]
+  [IsIrreducible k L M] [IsFaithful k L M] [IsTriangularizable k L M]
+
+lemma hasCentralRadical_and_of_isIrreducible_of_isFaithful :
+    HasCentralRadical k L ∧ (∀ x, x ∈ center k L ↔ toEnd k L M x ∈ k ∙ LinearMap.id) := by
+  have _i := nontrivial_of_isIrreducible k L M
+  obtain ⟨χ, hχ⟩ : ∃ χ : Module.Dual k (radical k L), Nontrivial (weightSpace M χ) :=
+    exists_nontrivial_weightSpace_of_isSolvable k (radical k L) M
```

Comment:
I expect that we could get the lemmas in this file to work over any principal ideal domain (of characteristic zero) instead of a field. The plan I see would be to define the nilradical of a Lie algebra (currently missing) and use it with `LieModule.exists_nontrivial_weightSpace_of_isNilpotent` instead of using the solvable radical with Lie's theorem as we are doing here. The conclusion of this lemma would then make a statement that the nilradical vanishes. We would prove elsewhere that vanishing nilradical is equivalent to `HasCentralRadical`.

Reply:
I think it is fine to leave it as future work. But please record your observation as a comment in the file.